AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/Mutations.dfy

```diff
@@ -90,6 +95,28 @@ module {:options "/functionSyntax:4" } Mutations {
     var kmsOperation: string;
     var success?: bool := false;
     var throwAwayError;
+    // TODO-HV-2-M3: Can this if condition also handle hv-2 item?
+    if (isTerminalHv2?) {
+      // TODO-HV-2-M4: Support other key manager strategy
+      :- Need(keyManagerStrategy.kmsSimple?, Types.KeyStoreAdminException(message:="only KMS Simple allow when mutating to hv-2."));
+      var decryptRes := KMSKeystoreOperations.DecryptKeyForHv1(
+        item,
+        KmsUtils.KmsSymmetricKeyArnToKMSConfiguration(Types.KmsSymmetricKeyArn.KmsKeyArn(item.KmsArn)),
```

Comment:
On re-reviewing it today. I found out that I am not considering MRK here. I will do it soon

Reply:
Good catch; we need to add some test cases to the test plan for this.